host/discoverd.go

```diff
@@ -105,3 +109,15 @@ func (d *DiscoverdManager) ConnectPeer(ips []string) error {
 	}
 	return err
 }
+
+func (d *DiscoverdManager) UpdateTags(tags map[string]string) error {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	for k, v := range tags {
+		d.inst.Meta[host.TagPrefix+k] = v
```

Comment:
This doesn't provide a method for removing tags. Perhaps the entire tag set should be provided?